examples/hotel_ads/add_performance_max_for_travel_goals_campaign.rb

```diff
@@ -0,0 +1,804 @@
+            hotel_image_asset.asset_field_type
+        )
+
+      # Keeps track of only required image assets. The service may sometimes
+      # suggest optional image assets.
+      unless required_image_asset_counts.has_key?(
+        hotel_image_asset.asset_field_type
+      )
+        next
+      end
+
+      required_image_asset_counts[hotel_image_asset.asset_field_type] += 1
+    end
+  end
+
+  # Adds more image assets to fulfill the requirements.
+  MIN_REQUIRED_IMAGE_ASSET_COUNTS.each do |asset_field_type, count|
+    next unless required_image_asset_counts[asset_field_type] < count
+
+    for i in required_image_asset_counts[asset_field_type]...count
+      default_uri = DEFAULT_IMAGE_ASSETS_INFO[asset_field_type][i]
+      puts "A default image URL '#{default_uri}' is used to create an image asset for the asset field type '#{asset_field_type}'."
+
+      operations +=
+        create_and_link_image_asset_operations(
+          client,
+          customer_id,
+          default_uri,
+          asset_field_type,
+          "Default image asset for the asset field type '%s'." %
+            asset_field_type
+        )
+    end
+  end
+
+  operations
+end
+
+# Creates a list of MutateOperations that create a new linked image asset.
+def create_and_link_image_asset_operations(
+  client,
+  customer_id,
+  url,
+  field_type,
+  asset_name
+)
+  operations = []
+  temp_id = next_temp_id
+
+  # Create the Image Asset.
+  operations << client.operation.mutate do |m|
+    m.asset_operation =
+      client.operation.create_resource.asset do |a|
+        a.resource_name = client.path.asset(customer_id, temp_id)
+        # Provide a unique friendly name to identify your asset.
+        # When there is an existing image asset with the same content but a different
+        # name, the new name will be dropped silently.
+        a.name = asset_name
+        a.type = :IMAGE
+        a.image_asset =
+          client.resource.image_asset do |image_asset|
+            image_asset.data = get_image_bytes(url)
+          end
+      end
+  end
+
+  # Create an AssetGroupAsset to link the Asset to the AssetGroup.
+  operations << client.operation.mutate do |m|
+    m.asset_group_asset_operation =
+      client.operation.create_resource.asset_group_asset do |aga|
+        aga.field_type = field_type
+        aga.asset_group =
+          client.path.asset_group(customer_id, ASSET_GROUP_TEMPORARY_ID)
+        aga.asset = client.path.asset(customer_id, temp_id)
+      end
+  end
+
+  operations
+end
+
+# Loads image data from a URL.
+def get_image_bytes(url)
+  URI.open(url).read
+end
+
+# Prints the details of a MutateGoogleAdsResponse.
+def print_response_details(response)
+  # Parse the mutate response to print details about the entities that
+  # were created by the request.
+  suffix = '_result'
+  response.mutate_operation_responses.each do |result|
+    result
+      .to_h
+      .select { |_k, v| v }
```

Comment:
I think this should be map instead of select, since there is no case where you're removing an element.

Reply:
I copied this one from another example but it does filter for the case where `v` is empty.
Without that it prints a lot more of created resources.

Reply:
Good point, I didn't consider the case that v would be empty. LGTM then.